source/llassetgen/source/DistanceTransform.cpp

```diff
@@ -1,300 +1,162 @@
-#include <algorithm>
-#include <limits>
-
-/*
- * If we sort png.h below the freetype includes, an compile error will be
- * generated warning about multiple includes of setjmp (on some version of
- * libpng). Because we don't use the setjmp functionality of freetype, this is
- * not applicable in this case. Unfortunately, the compile error can only be
- * fixed by editing png.h or reordering the includes.
- *
- * See https://sourceforge.net/p/enlightenment/mailman/message/12595025/ for
- * some discussion on this topic.
- */
-// clang-format off
-#include <png.h> // NOLINT
-#include <ft2build.h>
-#include FT_FREETYPE_H
-// clang-format on
+#include <iostream>
+#include <cassert>
+#include <cmath>
 
 #include <llassetgen/DistanceTransform.h>
 
-template <typename T>
-T square(T value) {
-    return value * value;
-}
-
-template <typename T>
-T clamp(T val, T min, T max) {
-    return std::max(min, std::min(max, val));
-}
-
 namespace llassetgen {
-    void DistanceTransform::resetInput(DimensionType _width, DimensionType _height, bool clear) {
-        assert(_width > 0 && _height > 0);
-        width = _width;
-        height = _height;
-        DimensionType size = (width * height + 7) / 8;
-        input.reset(new InputType[size]);
-        if (clear) memset(input.get(), 0, size);
-    }
-
-    bool DistanceTransform::inputAt(DimensionType offset) {
-        assert(offset < width * height && input.get());
-        return (input[offset / 8] >> (offset % 8)) & 1;
+    template<typename PixelType, bool flipped, bool invalidBounds>
+    PixelType DistanceTransform::getPixel(PositionType pos) {
+        const Image& image = std::is_same<PixelType, InputType>::value ? input : output;
```

Comment:
Is it idiomatic to return different data depending on the template type (in this case, pixels from different images)? I found it pretty confusing at first.

Reply:
The alternative is to use a even less obvious types like bool to switch between them.
Thus, I thought reading InputType / OutputType is better than just true / false.

Reply:
It should at least be documented. Also, it isn't needed in `setPixel` since AFAIK we never modify the input, only the output.

Reply:
An alternative would be having two functions `getInputPixel` and `getOutputPixel` which pass the correct image as an argument to `getPixel`, though I don't know if the additional function call would affect performance`.

Reply:
I combined them to avoid code redundancy

Reply:
You'd be adding a one-liner for `getInputPixel` and a two-liner for `getOutputPixel`. As far as redundancy is concerned, I think it's a negligible trade-off for the clearer interface.